packages/headless/src/features/recent-results/recent-results-slice.ts

```diff
@@ -0,0 +1,31 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  registerRecentResults,
+  clearRecentResults,
+  pushRecentResult,
+} from './recent-results-actions';
+import {getRecentResultsInitialState} from './recent-results-state';
+
+export const recentResultsReducer = createReducer(
+  getRecentResultsInitialState(),
+  (builder) => {
+    builder
+      .addCase(registerRecentResults, (state, action) => {
+        state.results = action.payload.results;
```

Comment:
We should ensure that `results` does not contain more values than the specified `maxLength`.

Reply:
Yeah good point I can slice it to the maxLength